pdf/pdfdocument.cpp

```diff
@@ -315,9 +318,17 @@ void PDFDocument::jobFinished(PDFJob *job)
 
 void PDFDocument::searchFinished(const QList<QPair<int, QRectF>> &matches)
 {
-    delete d->searchModel;
-    d->searchModel = new PDFSearchModel(matches);
-    emit searchModelChanged();
+    if (matches.count() != d->searchModel->count())
```

Comment:
I think it could make sense for simplicity to remove things that are not used. Would remove need to keep track of previous index etc if there thread didn't store any matches itself.